src/modules/select/wallets/1inch.ts

```diff
@@ -0,0 +1,38 @@
+import { mobileWalletInstallMessage } from '../content'
+import { WalletModule, Helpers, CommonWalletOptions } from '../../../interfaces'
+
+import oneInchIcon from '../wallet-icons/1inch.png'
+import oneInchIcon2x from '../wallet-icons/1inch@2x.png'
+
+function oneInchWallet(options: CommonWalletOptions): WalletModule {
+  const { preferred, label, iconSrc, svg } = options
+
+  return {
+    name: label || '1inch',
+    iconSrc: iconSrc || oneInchIcon,
+    iconSrcSet: iconSrc || oneInchIcon2x,
+    svg,
+    wallet: async (helpers: Helpers) => {
+      const { createModernProviderInterface, getProviderName } = helpers
+      const windowAsAny = window as any
+      const provider =
+        windowAsAny || (windowAsAny.web3 && windowAsAny.web3.currentProvider)
+
+      return {
+        provider,
+        interface:
+          provider && getProviderName(provider) === '1inch'
+            ? createModernProviderInterface(provider)
+            : null
+      }
+    },
+    type: 'injected',
+    link: `https://wallet.1inch.io?url=${window.location.host}`,
+    installMessage: mobileWalletInstallMessage,
+    mobile: true,
```

Comment:
@shoom3301 wanted to make a note here, after testing locally it seems this wallet can work on desktop as well with a little tweaking. If you would like to add please submit a new PR with `desktop: true` and an update to provider. thanks again!!